test/Main.purs

```diff
@@ -76,6 +78,23 @@ main = launchAff_ $ runSpec [consoleReporter] do
       quickCheck' 500 $ \(BadInput str) -> do
         Slug.parse str `assertEquals` Nothing
 
+  describe "Generate Slugs with Options" do
+    it "Generated slugs are idempotent" do
+      quickCheck' 500 \(Options' options) x -> do
+        let generate = Slug.generateWithOptions options
+            f = generate
+            g = generate >=> generate <<< Slug.toString
+        f x `assertEquals` g x
+
+  describe "Parse Slugs with Options" do
+    it "Slug parses successfully on valid input" do
+      quickCheck' 500 \(SlugWithOptions { slug, options }) -> do
+        Slug.parseWithOptions options (Slug.toString slug) `assertEquals` pure slug
```

Comment:
Both of these are currently failing, for example with:
```
Generate Slugs with Options
  ✗ Generated slugs are idempotent:

  (Just (Slug ⦖঍Ǐ鐵濟昒嚩씈�标ᜫ碉)) /= (Just (Slug ⦖঍Ǐ঍Ǐ鐵濟昒嚩씈�濟昒嚩঍Ǐ鐵濟昒嚩씈�标ᜫ碉))
  (Just (Slug ▄⮓饕ꭡ峀�撟᳕‬ᇟ根䵮)) /= (Just (Slug ▄⮓饕ꭡ峀�撟᳕‬ᇟ根᳕‬ᇟ饕ꭡ峀�撟᳕‬ᇟ根䵮))
  (Just (Slug ꣵ묛⍂떖犒꧑繜)) /= (Just (Slug ꣵ묛⍂떖犒꧑떖犒꧑繜))
Parse Slugs with Options
  ✗ Slug parses successfully on valid input:

  Nothing /= (Just (Slug 褬א᫓萈�ﻳ↹Ⅺ矻筷ꀊ))
  Nothing /= (Just (Slug 5啖衧㆑穦�纬勊穦�纬迏))
  Nothing /= (Just (Slug 쁦꺶衒⍄4꼼疀))
  Nothing /= (Just (Slug 섓鄷잲㦱ၩ쭀躩֤堿魌ꂞ㦱ၩ쭀躩֤堿))
  Nothing /= (Just (Slug 赚㚂䒪쏄䧀襒ढ़╦❉፣ጎ悷㼇))
  Nothing /= (Just (Slug ꨡ祡빰搅ᯝ苊ള郚怣ᕥ쨜빰搅ᯝ苊ള郚怣ᕥ릹觪))
```
I'll dig into why that's failing when I have a chance, but maybe I need to restrict `Options` to avoid edge cases here.

Reply:
So I think these properties don't hold if the `replacement` string contains a space _and_ another character that passes the `keepIf` condition. I just pushed some changes to restrict the `replacement` to dash or underscore, which passes the property tests, but I think the `Arbitrary` instance could reflect that constraint more accurately. I'll update it.

Reply:
Submitting for review as-is. I think it's reasonable to assume that if someone wants to use characters that would be filtered-out in the `replacement` string, then some properties go out the window. Happy to update the API if we want to make `replacement` more precisely-typed, but it is convenient to have it just be `String`.